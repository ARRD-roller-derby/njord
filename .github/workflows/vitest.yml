name: Vitest
on: push
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v2
      with:
        node-version: '16.x'
    - name: Install dependencies
      run: npm ci
    - name: Run Vitest tests
      run: npm run coverage
    - uses: actions/upload-artifact@v2
      if: always()
      with:
        name: vitest-report
        path: vitest-report/
        retention-days: 30