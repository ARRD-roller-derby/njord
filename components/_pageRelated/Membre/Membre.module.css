.container {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 15px;
  height: 100%;
}

.user {
  display: flex;
  display: flex;
  flex-direction: column;
  gap: 15px;
  width: 500px;
}

.avatarContainer {
  display: flex;
  justify-content: center;
  align-items: center;
}

.avatar {
  width: 50px;
}

@media screen and (max-width: 530px) {
  .user {
    width: 100%;
  }
}
