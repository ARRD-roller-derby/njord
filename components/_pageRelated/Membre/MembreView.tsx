import { UserInterface } from '../../../types/User.interface'
import UserBlockContact from '../../User/UserBlockContact/UserBlockContact'
import UserBlockDerby from '../../User/UserBlockDerby/UserBlockDerby'
import UserBlockIndentity from '../../User/UserBlockIndentity/UserBlockIndentity'
import UserBlockLeague from '../../User/UserBlockLeague/UserBlockLeague'
import AuthentificatedLayout from '../../_layouts/Authentificated/Authentificated'
import classes from './Membre.module.css'

interface props {
  readonly user: UserInterface
  readonly reSync: Function
  readonly uri: string
}

export default function MembreView({ user, reSync, uri }: props) {
  return (
    <AuthentificatedLayout>
      <div className={classes.container}>
      {user ? <div className={classes.user}>
        <UserBlockIndentity user={user} uri={uri} reSync={reSync} />
        <UserBlockContact user={user} uri={uri} reSync={reSync} />
        <UserBlockDerby user={user} uri={uri} reSync={reSync} />
        <UserBlockLeague user={user} uri={uri} reSync={reSync} />
        </div>:<>Utilisateur inconnu</>}

      </div>
    </AuthentificatedLayout>
  )
}
