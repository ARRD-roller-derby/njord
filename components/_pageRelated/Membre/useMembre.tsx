import { useSession } from 'next-auth/react'
import { useState, useEffect } from 'react'
import { UserInterface } from '../../../types/User.interface'
import useDBSync from '../../_hooks/useDBSync'
import { indexDB } from "../../../db/indexDB.connect";

export default function useMembre(id:string) {
  const { data: session } = useSession(),
    { data: users, reSync } = useDBSync('league/users', 'users'),
    [user, setUser] = useState<UserInterface>(),
    uri = '/users/updateField'

  useEffect(() => {
    if (session) getUser()
  }, [users, session])

  async function getUser() {
    if (!session.user?.league?.id) return
    const userDb = await indexDB.users.where({ _id: id }).first()
    setUser(userDb)
  }

  return { reSync, user, uri,isMe: user ? session.user._id === user?._id:false }
}
