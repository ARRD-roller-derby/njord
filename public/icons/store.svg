<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg
   xmlns:dc="http://purl.org/dc/elements/1.1/"
   xmlns:cc="http://creativecommons.org/ns#"
   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
   xmlns:svg="http://www.w3.org/2000/svg"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
   viewBox="0 0 576 512"
   version="1.1"
   id="svg10"
   sodipodi:docname="store-duotone.svg"
   inkscape:version="1.0.1 (3bc2e813f5, 2020-09-07)">
  <metadata
     id="metadata14">
    <rdf:RDF>
      <cc:Work
         rdf:about="">
        <dc:format>image/svg+xml</dc:format>
        <dc:type
           rdf:resource="http://purl.org/dc/dcmitype/StillImage" />
      </cc:Work>
    </rdf:RDF>
  </metadata>
  <sodipodi:namedview
     pagecolor="#ffffff"
     bordercolor="#666666"
     borderopacity="1"
     objecttolerance="10"
     gridtolerance="10"
     guidetolerance="10"
     inkscape:pageopacity="0"
     inkscape:pageshadow="2"
     inkscape:window-width="1920"
     inkscape:window-height="1009"
     id="namedview12"
     showgrid="false"
     inkscape:zoom="0.83984375"
     inkscape:cx="329.78442"
     inkscape:cy="254.52185"
     inkscape:window-x="1272"
     inkscape:window-y="-8"
     inkscape:window-maximized="1"
     inkscape:current-layer="svg10" />
  <!--! Font Awesome Pro 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) Copyright 2022 Fonticons, Inc. -->
  <defs
     id="defs4">
    <style
       id="style2">.fa-secondary{opacity:.4}</style>
  </defs>
  <path
     class="fa-primary"
     d="M490.3 13.13L547.6 103.8C577.3 150.7 551 215.8 495.5 223.2C491.6 223.7 487.6 224 483.4 224C457.4 224 434.2 212.6 418.3 195C402.4 212.6 379.2 224 353.1 224C327 224 303.8 212.6 287.9 195C272 212.6 248.9 224 222.7 224C196.7 224 173.5 212.6 157.6 195C141.7 212.6 118.5 224 92.36 224C88.3 224 84.21 223.7 80.24 223.2C24.92 215.8-1.255 150.6 28.33 103.8L85.66 13.13C90.76 4.979 99.87 0 109.6 0H466.4C476.1 0 485.2 4.978 490.3 13.13H490.3z"
     id="path6"
     style="fill:#999999" />
  <path
     class="fa-secondary"
     d="M80.23 223.2C84.21 223.7 88.3 224 92.36 224C105.1 224 117.2 221.3 128 216.5V384H448V216.5C458.8 221.3 470.8 224 483.4 224C487.6 224 491.6 223.7 495.5 223.2C501.4 222.4 506.9 221 512 219.1V448C512 483.3 483.3 512 448 512H128C92.65 512 64 483.3 64 448V219.1C69.08 221 74.5 222.4 80.24 223.2H80.23z"
     id="path8"
     style="fill:#b3b3b3" />
</svg>
