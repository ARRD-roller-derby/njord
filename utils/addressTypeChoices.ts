import { addressType } from "../types/addressType.enum";

export const addressTypeChoices = [
  {
    label: 'Mon adresse',
    value:addressType.personnal
  },
  {
    label: 'Stade',
    value:addressType.stadium
  },
  {
    label: 'Point de covoiturage',
    value:addressType.carpool
  },
  {
    label: 'Autre',
    value:addressType.other
  }
]