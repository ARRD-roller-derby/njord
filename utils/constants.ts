export const DISCORD_NEWS_HOOK = process.env.DISCORD_NEWS_HOOK || ''