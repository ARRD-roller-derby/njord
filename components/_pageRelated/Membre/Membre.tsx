import MembreView from "./MembreView";
import useMembre from './useMembre';

interface props {
  readonly id: string
}

export default function Membre({id}:props) {
  const props = useMembre(id);

  return <MembreView {...props}/>
}
