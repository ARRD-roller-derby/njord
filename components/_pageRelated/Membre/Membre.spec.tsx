import { afterEach, describe, expect, it, vi } from 'vitest'
import { cleanup } from '@testing-library/react'
import Membre from './Membre'
import { render, screen } from '../../../utils/test-utils'
import { SocketContext } from '../../../stores/socket.store'

describe('<Membre />', () => {
  afterEach(cleanup)

  it('Check snapshot', async () => {

    const { asFragment } = render(
      <SocketContext.Provider value={[null, vi.fn]}>
        <Membre id={'id'} />
      </SocketContext.Provider>
    )
    expect(await screen.findByText('200 Dr.')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })
})
