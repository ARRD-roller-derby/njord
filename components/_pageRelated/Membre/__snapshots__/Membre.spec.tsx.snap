// Vitest Snapshot v1

exports[`<Membre /> > Check snapshot 1`] = `
<DocumentFragment>
  <div
    class="_container_lmcue_1"
    data-color-mode="dark"
  >
    <header
      class="_header_lmcue_18"
    >
      <header
        class="_header_1yhnh_1"
      >
        <div
          class="_appName_1yhnh_7"
        >
          <a
            class="_title_1045q_1"
            title="retour à l'accueil"
          >
            Njörd
          </a>
          <div
            class="_link_1ld19_1"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="flêche"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <div>
              Njörd
            </div>
          </div>
        </div>
        <div
          class="_appName_1yhnh_7"
        >
          <div
            class="_count_snjyq_1"
          >
            200 Dr.
          </div>
          <a
            class="_container_1ssp7_1"
            data-active="false"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="clôche"
                class="_bell_1ssp7_11"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <div
              class="_count_13krl_1"
            >
               10
            </div>
          </a>
          <div
            class="_container_1cugo_1"
          >
            <div
              class="_avatar_1cugo_12"
              data-ignore="true"
              data-testid="avatar"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2730%27%20height=%2730%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="/public/static/images/profile.webp"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
          </div>
        </div>
      </header>
      <div
        class="Toastify"
      />
    </header>
    <div
      class="_menu_lmcue_22"
    />
    <main
      class="_main_lmcue_12"
    >
      <div
        class="_box_lmcue_31"
      >
        <div
          class="_container_9xohe_1"
        >
          Utilisateur inconnu
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
